### PR TITLE
Rename some var names in replication stream logic

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Rename source cluster key to client cluster key
* Rename target cluster key to server cluster key

**Why?**
Before this PR:
source cluster key means the source of the stream, but this usually means the target of replication data flow;
target cluster key means the target of the stream, but this usually means the source of replication data flow;
the meaning can be misleading

**How did you test it?**
N/A

**Potential risks**
N/A

**Is hotfix candidate?**
N/A

### CODE REVIEW

**Comment by wxing1292** [comment]
review #4178 first

File: client/history/metadata.go [hunk 1]
```diff
@@ -51,66 +51,63 @@ func EncodeClusterShardMD(
...
+	metadataValues := clusterShardMD.Get(metadataKey)
+	if len(metadataValues) != 1 {
+		return 0, serviceerror.NewInvalidArgument(fmt.Sprintf(
+			"unable to parse metadata key %v: %v",
+			metadataKey,
+			metadataValues,
 		))
 	}
-	shardID, err = strconv.Atoi(shardIDs[0])
+	metadataValue, err := strconv.Atoi(metadataValues[0])
```
Comment (hehaifengcn) [inline client/history/metadata.go:104]: use `strconv.ParseInt(.., 10, 32)` instead of casting to int32 explicitly.

**Review by hehaifengcn — APPROVED** [review]

File: client/history/metadata.go (hunk 1 above)
Comment (wxing1292) [reply client/history/metadata.go:104]: according to the doc, "Atoi is equivalent to ParseInt(s, 10, 0)" so what is the difference? 

strconv.ParseInt(.., 10, 32) will return int64
strconv.Atoi will return int

File: service/history/replication/grpc_stream_client.go [hunk 2]
```diff
@@ -26,46 +26,69 @@ package replication
...
-			ClusterName: targetShardKey.ClusterName,
-			ShardID:     targetShardKey.ShardID,
+			ClusterID: serverShardKey.ClusterID,
+			ShardID:   serverShardKey.ShardID,
 		},
 	))
 	return adminClient.StreamWorkflowReplicationMessages(ctx)
 }
+
+func clusterIDToClusterName(
```
Comment (yux0) [inline service/history/replication/grpc_stream_client.go:84]: can this method and clusterIDToClusterNameShardCount combine and make it a util?

**Review by yux0 — APPROVED** [review]

File: service/history/replication/grpc_stream_client.go (hunk 2 above)
Comment (wxing1292) [reply service/history/replication/grpc_stream_client.go:84]: it definitely can, will do in later PRs

**Pull request merged**